app/assets/javascripts/govuk_publishing_components/load-analytics.js

```diff
@@ -0,0 +1,261 @@
+    window.GOVUK.analyticsVars = window.GOVUK.analyticsVars || {}
+    window.GOVUK.analyticsVars.gaProperty = 'UA-UNSET'
+    window.GOVUK.analyticsVars.gaPropertyCrossDomain = 'UA-145652997-1'
+    window.GOVUK.analyticsVars.primaryLinkedDomains = ['account.gov.uk']
+    window.GOVUK.analyticsVars.linkedDomains = linkedDomains
+
+    if (this.arrayContains(currentDomain, this.productionDomains)) {
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-1'
+    } else if (this.arrayContains(currentDomain, this.stagingDomains)) {
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-20'
+    } else if (this.arrayContains(currentDomain, this.integrationDomains)) {
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-22'
+    } else {
+      window.GOVUK.analyticsVars.gaPropertyCrossDomain = 'UA-UNSET'
+    }
+
+    // Load universal analytics
+    if (typeof window.GOVUK.analyticsInit !== 'undefined') {
+      window.GOVUK.analyticsInit()
+    }
+  },
+
+  loadGa4: function (currentDomain) {
+    currentDomain = currentDomain || window.location.hostname
+    var environment = ''
+
+    // Categorise current environment and set a couple UA variables
+    if (this.arrayContains(currentDomain, this.productionDomains)) {
+      environment = 'production'
+    } else if (this.arrayContains(currentDomain, this.stagingDomains)) {
+      environment = 'staging'
+    } else if (this.arrayContains(currentDomain, this.integrationDomains)) {
+      environment = 'integration'
+    } else if (this.arrayContains(currentDomain, this.developmentDomains) || currentDomain.indexOf('.dev.gov.uk') !== -1) {
+      environment = 'development'
+    }
+
+    // If we recognise the environment (i.e. the string isn't empty), load in GA4
+    if (environment) {
+      // If analytics-ga4.js exists and our detected environment has 'initialiseGA4' set to true, load GA4.
+      if (typeof window.GOVUK.analyticsGa4.init !== 'undefined' && this.ga4EnvironmentVariables[environment].initialiseGA4) {
+        window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+        window.GOVUK.analyticsGa4.vars = window.GOVUK.analyticsGa4.vars || {}
+        window.GOVUK.analyticsGa4.vars.id = this.ga4EnvironmentVariables.id
+        window.GOVUK.analyticsGa4.vars.auth = this.ga4EnvironmentVariables[environment].auth
+        window.GOVUK.analyticsGa4.vars.preview = this.ga4EnvironmentVariables[environment].preview
+        window.GOVUK.analyticsGa4.vars.environment = environment // Used for testing and debugging
+
+        var gemVersion = document.querySelector('meta[name="govuk:components_gem_version"]')
+        if (gemVersion) {
+          gemVersion = gemVersion.getAttribute('content')
+        } else {
+          gemVersion = 'not found'
+        }
+
+        window.GOVUK.analyticsGa4.vars.gem_version = gemVersion
+        window.GOVUK.analyticsGa4.init()
+      }
+    }
+  },
+
+  arrayContains: function (valueToFind, array) {
```

Comment:
Do you think it is pointless having this function? It might be more readable if I just did

`if(productionDomains.indexOf(currentDomain) !== 1)` etc. in the code.

Reply:
I think it's okay - it makes the code more readable. I was wondering if it could be renamed to be something even clearer, like maybe `if (theEnvironmentIs(this.staging, currentDomain))` but that might not be necessary.